### PR TITLE
11969 - Border Outline trait. What?!?! Oh no it's a new trait! FLEE FOR THE HILLS, HAIR ON FIRE!!!

### PR DESCRIPTION
**Pull request by Cattlesquat**
BorderOutline trait makes a colored border around the piece "possibly contingent on a property" 

image

image

### CODE REVIEW

**Comment by BrentEaston** [comment]
So this is sort of a per-piece configurable selection highlighter that doesn't only work when selected?

**Comment by Cattlesquat** [comment]
@BrentEaston yes exactly -- choice between "works always" (no property specified) or "works when property is true"

File: vassal-app/src/main/java/VASSAL/counters/BorderOutline.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+  @Override
+  public Shape getShape() {
+    return piece.getShape();
+  }
+
+  @Override
+  public Rectangle boundingBox() {
+    final Rectangle r = piece.boundingBox();
+    r.add(piece.boundingBox());
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/counters/BorderOutline.java:105]: Why are you adding the bounding box when `r` starts out as the bounding box?

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/counters/BorderOutline.java]: Because of poor copying and pasting I guess :) All fixed

**Pull request merged**